semanticsearch/README.md

```diff
@@ -0,0 +1,440 @@
+
+To recreate the document corpus:
+
+1. Download the following files:
+    * ftp://ftp.ncdc.noaa.gov/pub/data/ghcn/daily/by_year/2014.csv.gz
+    * ftp://ftp.ncdc.noaa.gov/pub/data/ghcn/daily/by_year/2015.csv.gz
+    * ftp://ftp.ncdc.noaa.gov/pub/data/ghcn/daily/by_year/2016.csv.gz
+    * ftp://ftp.ncdc.noaa.gov/pub/data/ghcn/daily/ghcnd-stations.txt
+    * ftp://ftp.ncdc.noaa.gov/pub/data/ghcn/daily/ghcnd-countries.txt
+    * ftp://ftp.ncdc.noaa.gov/pub/data/ghcn/daily/ghcnd-states.txt
+2. Decompress measurement files. For example: `gunzip 2016.csv.gz`
+3. Sort the files by station. For example: `sort --field-separator=',' --key=1,2 -o 2016-sorted.csv 2016.csv`
+4. Execute a script like `_tools/process.py` to create json documents.
+5. Make sure that the JSON documents are randomly ordered. (The script orders measurements of the same station next to each other). This can be achieved with `shuf documents.json > documents1.json`.
+6. Compress the documents json file: `bzip2 -9 -c documents1.json > documents.json.bz2`
+
+### Example Document
+
+```json
+{
+  "date": "2016-01-01T00:00:00",
+  "TAVG": 22.9,
+  "station": {
+    "elevation": 34.0,
+    "name": "SHARJAH INTER. AIRP",
+    "country": "United",
+    "gsn_flag": "GSN",
+    "location": {
+      "lat": 25.333,
+      "lon": 55.517
+    },
+    "country_code": "AE",
+    "wmo_id": "41196",
+    "id": "AE000041196"
+  },
+  "TMIN": 15.5
+}
+```
+
+## Parameters
+
+This workload allows the following parameters to be specified using `--workload-params`:
+
+* `bulk_size` (default: 5000)
+* `bulk_indexing_clients` (default: 8): Number of clients that issue bulk indexing requests.
+* `number_of_replicas` (default: 0)
+* `number_of_shards` (default: 1)
+* `max_num_segments` : Number segments that used for force merge, force merge is skipped if parameter has not been set
+* `index_settings`: A list of index settings. Index settings defined elsewhere (e.g. `number_of_replicas`) need to be overridden explicitly.
+* `cluster_health` (default: "green"): The minimum required cluster health.
+* `target_throughput` (default: default values for each operation): Number of requests per second, `none` for no limit.
+* `search_clients`: Number of clients that issues search requests.
+* `concurrent_segment_search_enabled` (default: "false"): Setting enables or disables consurrent segment search feature for the cluster. Setting is avaliable starting from version 2.12.
+
+### Running a benchmark
+
+Before running a benchmark, ensure that the load generation host is able to access your cluster endpoint and that the 
+appropriate dataset is available on the host.
+
+Currently, we support 5 test procedures for the semantic search workload. The default procedure is `hybrid-query-aggs-light` and does create an index, ingest data and run a base set of search queries.
+
+To run the default workload, invoke the following command. In that example we set number some extra parameters directly as part of the command.
```

Comment:
Nit: "we set number some extra parameters" --> "we set some extra parameters"

Reply:
ack